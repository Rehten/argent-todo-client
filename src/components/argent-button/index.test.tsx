import React from "react";
import { ArgentButton } from "./index";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import ReactTestUtils from 'react-dom/test-utils';
import TestRenderer from "react-test-renderer";


describe('ArgentButton', () => {
    let container: HTMLElement;

    beforeEach(() => {
        // подготавливаем DOM-элемент, куда будем рендерить
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        // подчищаем после завершения
        unmountComponentAtNode(container);
        container.remove();
        container = (null as any) as HTMLElement;
    });

    it('should be object', () => {
        expect(ArgentButton).toBeDefined();
        expect(ArgentButton).not.toBeNull();
        expect(ArgentButton).toBeInstanceOf(Object);
    });

    it('should container HTML button node', () => {
        act(() => {
            render(<ArgentButton />, container);
        });

        const htmlButton: HTMLButtonElement | null = container.querySelector('button');
        expect(htmlButton).not.toBeNull();
    });

    it('should have div wrapper with data-test id', () => {
        const testMarker: string = 'Qwerty';

        act(() => {
            render(<ArgentButton dataTest={testMarker} />, container);
        });

        const divWrapper: HTMLDivElement | null = container.querySelector(`div[data-test]`);

        expect(divWrapper).not.toBeNull();
        expect(divWrapper?.dataset.test).toEqual(testMarker);
    });

    it('should call props.onClick after user click', () => {
        const onClick = jest.fn();

        act(() => {
            render(<ArgentButton onClick={onClick} />, container);
        });

        const btnRef: HTMLButtonElement = container.querySelector('button') as HTMLButtonElement;

        ReactTestUtils.Simulate.click(btnRef);

        expect(onClick).toHaveBeenCalled();
    });

    it('should be same as snapshot', () => {
        const rslt = TestRenderer.create(<ArgentButton dataTest="Snapshot Testing" variant="outline-danger" size="lg" active />);
        expect(rslt).toMatchSnapshot();
    });
});