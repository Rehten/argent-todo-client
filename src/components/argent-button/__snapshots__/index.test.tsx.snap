// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArgentButton should be same as snapshot 1`] = `
<div
  data-test="Snapshot Testing"
>
  <button
    className="btn active btn-outline-danger btn-lg"
    disabled={false}
    type="button"
  />
</div>
`;
